packages/web-app-files/tests/unit/components/AppBar/SelectedResources/SizeInfo.spec.js

```diff
@@ -0,0 +1,114 @@
+import { mount, createLocalVue } from '@vue/test-utils'
+import Vuex from 'vuex'
+import GetTextPlugin from 'vue-gettext'
+import DesignSystem from 'owncloud-design-system'
+import stubs from 'tests/unit/stubs'
+import SizeInfo from '../../../../../src/components/AppBar/SelectedResources/SizeInfo.vue'
+
+const localVue = createLocalVue()
+localVue.use(Vuex)
+localVue.use(DesignSystem)
+localVue.use(GetTextPlugin, {
+  translations: 'translation.json',
+  silent: true
+})
+
+const selectedFiles = [
+  {
+    path: '/folder',
+    size: '5'
+  },
+  {
+    path: '/lorem.txt',
+    size: '100'
+  },
+  {
+    path: '/sample.pdf',
+    size: '200'
+  }
+]
+const componentStubs = { ...stubs, translate: true }
+
+describe('SizeInfo component', () => {
+  afterEach(() => {
+    jest.clearAllMocks()
+  })
+
+  jest
+    .spyOn(SizeInfo.mixins[0].methods, 'getResourceSize')
+    .mockImplementation(size => (isNaN(size) ? '?' : size + ' B'))
```

Comment:
I don't feel good about having extra logic in the mock. Can we have different mocks for different cases? That would make the tests easier to read and have less risk that the logic in the tests break